gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -267,4 +218,124 @@ private Instant getCutoffTimeForRef(String reference, Map<String, Instant> dropp
                 GCUtil.deserializeReference(reference).getName(),
                 gcParams.getDefaultCutOffTimestamp());
   }
+
+  // --- Methods for computing expired content rows ----------
+  /**
+   * convert scala Iterator of reference into scala Iterator of {@link Row}.
+   *
+   * <p>Each reference can produce one or more rows.
+   */
+  private scala.collection.Iterator<Row> getExpiredContentRows(
+      scala.collection.Iterator<String> references,
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap,
+      String runId,
+      Timestamp startedAt) {
+    NessieApiV1 api = GCUtil.getApi(gcParams.getNessieClientConfigs());
+    TaskContext.get()
+        .addTaskCompletionListener(
+            context -> {
+              LOGGER.info("Closing the nessie api for compute expired contents task");
+              api.close();
+            });
+    List<Iterator<Row>> iterators = new ArrayList<>();
```

Comment:
This can all be replaced with this one - no need to collect + re-stream.
```
    return references.flatMap(
        reference ->
            JavaConverters.asScalaIterator(
                    walkAllCommitsInReference(
                        api,
                        GCUtil.deserializeReference(reference),
                        liveContentsBloomFilterMap,
                        runId,
                        startedAt))
                .toTraversable());
```

Reply:
thanks. Replaced.